dashboard.py

```diff
@@ -0,0 +1,274 @@
+    else:
+        fig = px.histogram(df,
+                        x = x_var,
+                        color = color_by,
+                        color_discrete_sequence = px.colors.qualitative.Bold).update_xaxes(categoryorder = sort_by)
+    
+    fig.update_layout(title = {'text': f'Distribution of {x_var} Colored by {color_by}'})
+
+    return fig
+
+# Pie Section
+
+@app.callback(
+    #pie output
+    Output(component_id='pie-plot', component_property='figure'),
+    #pie input (var)
+    Input(component_id='prct-brkdwn', component_property='value')
+)
+
+def make_pie_plot(var):
+    #generate pie chart
+    if(var == 'Subspecies'):
+        pie_fig = px.pie(df,
+                 names = var,
+                 color_discrete_sequence = px.colors.qualitative.Bold,
+                 hover_data = ['Species'])
+    else:
+        pie_fig = px.pie(df,
+                 names = var,
+                 color_discrete_sequence = px.colors.qualitative.Bold)
+        pie_fig.update_traces(textposition = 'inside', textinfo = 'percent+label')
+    
+    pie_fig.update_layout(title = {'text': f'Percentage Breakdown of {var}'})
+
+    return pie_fig
+
+# Image Section
+
+# Callback for Image Species Selection
+@app.callback(
+    Output(component_id = 'subspecies-show', component_property= 'options'),
+    Input(component_id = 'species-show', component_property = 'value')
+)
+
+def set_subspecies_options(selected_species):
+    # Set subspecies options based on selected species
+    return [{'label': i, 'value': i} for i in all_species[selected_species]]
+
+# Callback for Image Subspecies Selection
+@app.callback(
+    Output(component_id = 'subspecies-show', component_property= 'value'),
+    Input(component_id = 'subspecies-show', component_property = 'options')
+)
+
+def set_subspecies_value(available_options):
+    # Collect selected subspecies
+    return available_options[0]['value']
+
+@app.callback(
+    Output('image-1', 'children'),
+    Input('display-img', 'n_clicks'),
+    #State('species-show', 'value'),
+    State('subspecies-show', 'value'),
+    State('which-view', 'value'),
+    State('which-sex', 'value'),
+    State('hybrid?', 'value'),
+    prevent_initial_call = True
+)
+
+# Retrieve a single image
+def get_image(n_clicks, subspecies, view, sex, hybrid):
+    #if n_clicks is None:
+    #    raise PreventUpdate
+   # else:
+    filename = get_filename(subspecies, view, sex, hybrid)
+    if filename == 0:
+        #print("No Such Images. Please make another selection.")
+        return html.H4("No Such Images. Please make another selection.", 
+                    style = {"color":"MidnightBlue"})
+    if 'D_lowres' in filename:
+        #remove 'tif' from filename and replace with 'png' in url
+        image_directory = "https://github.com/Imageomics/dashboard-prototype/blob/feature/image-options/test_data/images/dorsal_images/" + filename[:-3] + "png?raw=true"
```

Comment:
The only difference between this line and line 248 seems to be “dorsal” vs “ventral” in the URL path, so it seems the code redundancy could be limited to setting just that part in the conditional clause. And `image_directory` is really a path not a directory, right? So maybe name it `image_path` for clarity?